### ISSUE TITLE
[external bug report][from claude] x * sigmoid(x) wrong results when x is computed intermediate

### ISSUE THREAD

**Issue by zoecarver** [tenstorrent/tt-lang#443]
## Description

`x * sigmoid(x)` (SiLU pattern) produces incorrect results when `x` is a computed intermediate value within a `with` block, but works correctly when `x` comes directly from a DFB `wait()`.

## Observed behavior

| Case | Code | PCC vs PyTorch |
|------|------|----------------|
| BROKEN | `y = av + bv; o.store(y * ttl.math.sigmoid(y))` | **0.665** |
| FIXED (workaround) | store `y` to intermediate DFB, read back, then `o.store(tv * ttl.math.sigmoid(tv))` | **0.99999** |
| WORKS | `o.store(xv * ttl.math.sigmoid(xv))` where `xv` is from DFB wait | **0.99999** |

The key difference: when the value passed to `sigmoid` is a **computed expression result** (like `av + bv`), the multiplication `y * sigmoid(y)` appears to use a stale/clobbered value for the left-hand `y`. When `y` comes directly from a DFB wait (i.e., loaded from a CB), it works correctly.

## Reproducer

Three kernels that all compute `silu(a + b)`:

```python
import torch
import ttnn
import ttl

TILE = 32

def pcc(a, b):
    a, b = a.float().flatten(), b.float().flatten()
    return torch.corrcoef(torch.stack([a, b]))[0, 1].item()

# --- BROKEN: SiLU on computed intermediate ---
@ttl.kernel(grid=(1, 1))
def silu_on_computed_BROKEN(a, b, out):
    tiles = a.shape[0] // TILE
    a_dfb = ttl.make_dataflow_buffer_like(a, shape=(1, 1), buffer_factor=2)
    b_dfb = ttl.make_dataflow_buffer_like(b, shape=(1, 1), buffer_factor=2)
    out_dfb = ttl.make_dataflow_buffer_like(out, shape=(1, 1), buffer_factor=2)

    @ttl.compute()
    def compute():
        for t in range(tiles):
            with a_dfb.wait() as av, b_dfb.wait() as bv, out_dfb.reserve() as o:
                y = av + bv
                o.store(y * ttl.math.sigmoid(y))

    @ttl.datamovement()
    def dm_read():
        for t in range(tiles):
            with a_dfb.reserve() as blk:
                tx = ttl.copy(a[t, 0], blk); tx.wait()
            with b_dfb.reserve() as blk:
                tx = ttl.copy(b[t, 0], blk); tx.wait()

    @ttl.datamovement()
    def dm_write():
        for t in range(tiles):
            with out_dfb.wait() as blk:
                tx = ttl.copy(blk, out[t, 0]); tx.wait()

# --- FIXED: intermediate DFB workaround ---
@ttl.kernel(grid=(1, 1))
def silu_on_computed_FIXED(a, b, out):
    tiles = a.shape[0] // TILE
    a_dfb = ttl.make_dataflow_buffer_like(a, shape=(1, 1), buffer_factor=2)
    b_dfb = ttl.make_dataflow_buffer_like(b, shape=(1, 1), buffer_factor=2)
    tmp_dfb = ttl.make_dataflow_buffer_like(a, shape=(1, 1), buffer_factor=2)
    out_dfb = ttl.make_dataflow_buffer_like(out, shape=(1, 1), buffer_factor=2)

    @ttl.compute()
    def compute():
        for t in range(tiles):
            with a_dfb.wait() as av, b_dfb.wait() as bv, tmp_dfb.reserve() as tmp:
                tmp.store(av + bv)
            with tmp_dfb.wait() as tv, out_dfb.reserve() as o:
                o.store(tv * ttl.math.sigmoid(tv))

    @ttl.datamovement()
    def dm_read():
        for t in range(tiles):
            with a_dfb.reserve() as blk:
                tx = ttl.copy(a[t, 0], blk); tx.wait()
            with b_dfb.reserve() as blk:
                tx = ttl.copy(b[t, 0], blk); tx.wait()

    @ttl.datamovement()
    def dm_write():
        for t in range(tiles):
            with out_dfb.wait() as blk:
                tx = ttl.copy(blk, out[t, 0]); tx.wait()

# --- WORKS: SiLU on direct DFB value ---
@ttl.kernel(grid=(1, 1))
def silu_on_dfb_WORKS(x, out):
    tiles = x.shape[0] // TILE
    x_dfb = ttl.make_dataflow_buffer_like(x, shape=(1, 1), buffer_factor=2)
    out_dfb = ttl.make_dataflow_buffer_like(out, shape=(1, 1), buffer_factor=2)

    @ttl.compute()
    def compute():
        for t in range(tiles):
            with x_dfb.wait() as xv, out_dfb.reserve() as o:
                o.store(xv * ttl.math.sigmoid(xv))

    @ttl.datamovement()
    def dm_read():
        for t in range(tiles):
            with x_dfb.reserve() as blk:
                tx = ttl.copy(x[t, 0], blk); tx.wait()

    @ttl.datamovement()
    def dm_write():
        for t in range(tiles):
            with out_dfb.wait() as blk:
                tx = ttl.copy(blk, out[t, 0]); tx.wait()

# --- Test harness ---
device = ttnn.open_device(device_id=0, l1_small_size=32768)
DRAM = ttnn.DRAM_MEMORY_CONFIG

torch.manual_seed(42)
N = 128
a_t = torch.randn(N, TILE, dtype=torch.bfloat16)
b_t = torch.randn(N, TILE, dtype=torch.bfloat16)
sum_ab = (a_t.float() + b_t.float())
ref_silu = sum_ab * torch.sigmoid(sum_ab)

def to_dev(t):
    return ttnn.from_torch(t, ttnn.bfloat16, layout=ttnn.TILE_LAYOUT,
                           device=device, memory_config=DRAM)

a_tt, b_tt = to_dev(a_t), to_dev(b_t)

out1 = to_dev(torch.zeros_like(a_t))
silu_on_computed_BROKEN(a_tt, b_tt, out1)
r1 = ttnn.to_torch(out1).reshape(N, TILE).float()
print(f"BROKEN (computed intermediate): PCC={pcc(ref_silu, r1):.6f}")  # ~0.665

out2 = to_dev(torch.zeros_like(a_t))
silu_on_computed_FIXED(a_tt, b_tt, out2)
r2 = ttnn.to_torch(out2).reshape(N, TILE).float()
print(f"FIXED  (intermediate DFB):      PCC={pcc(ref_silu, r2):.6f}")  # ~0.99999

sum_tt = to_dev(sum_ab.to(torch.bfloat16))
out3 = to_dev(torch.zeros_like(a_t))
silu_on_dfb_WORKS(sum_tt, out3)
r3 = ttnn.to_torch(out3).reshape(N, TILE).float()
print(f"WORKS  (direct DFB value):      PCC={pcc(ref_silu, r3):.6f}")  # ~0.99999

ttnn.close_device(device)
```

## Expected output
All three cases should produce PCC > 0.999.

## Actual output
```
BROKEN (computed intermediate): PCC=0.664688
FIXED  (intermediate DFB):      PCC=0.999995
WORKS  (direct DFB value):      PCC=0.999994
```

## Hypothesis

The DST register holding the computed intermediate `y` gets clobbered by the `sigmoid` computation, so the left-hand side of `y * sigmoid(y)` reads stale data. When `y` comes from a DFB wait (CB read), the hardware reloads it correctly for the multiply.

## Environment
- Blackhole hardware
- Tested on real device (not simulator)

**Comment by zoecarver** [comment]
Fix:
```diff
diff --git a/lib/Dialect/TTL/Transforms/ConvertTTLTileOpsToTTKernel.cpp b/lib/Dialect/TTL/Transforms/ConvertTTLTileOpsToTTKernel.cpp
index 98a54960..bf86cc30 100644
--- a/lib/Dialect/TTL/Transforms/ConvertTTLTileOpsToTTKernel.cpp
+++ b/lib/Dialect/TTL/Transforms/ConvertTTLTileOpsToTTKernel.cpp
@@ -521,8 +521,8 @@ struct TTLCopyDstToTTKernel : OpConversionPattern<CopyDstOp> {
     Value srcIdx = arith::ConstantIndexOp::create(rewriter, loc, *srcDstIdx);
     Value dstIdx = arith::ConstantIndexOp::create(rewriter, loc, dstDstIdx);
 
-    // Emit copy_dest_values(dst0, dst1): copies DST[dst1] → DST[dst0].
-    ttk::CopyDestValuesOp::create(rewriter, loc, dstIdx, srcIdx);
+    // Emit copy_dest_values(old, new): copies DST[old] -> DST[new].
+    ttk::CopyDestValuesOp::create(rewriter, loc, srcIdx, dstIdx);
```
